tensorflow/compiler/xla/service/gpu/cudnn_fused_conv_rewriter.cc

```diff
@@ -330,46 +330,77 @@ StatusOr<bool> FuseConvAlpha(HloComputation* comp) {
 
 // The format of the serialized graph describing a sequence of ops fused
 // into the cuDNN convolution Custom Call is
-// "UID:[output_type]conv();UID[output_type]:op_name({operand
-// UIDs});UID:[output_type]op_name({operands UIDs});..." with the convolution
-// assumed to be the first op in the graph. Currently, multiplication and
-// division by a broadcast scalar, addition of a matrix bias, the application of
-// a ReLU activation and the calculation of the maximum of the absolute value
-// are supported.
+// "UID:[output_type]conv();UID[output_type]:op_name(operand
+// UID);UID:[output_type]op_name(operand UID);..." with the convolution assumed
+// to be the first op in the graph. Operand UIDs identifying ops outside the
+// serialized graph are elided. Currently, multiplication and division by a
+// broadcast scalar, addition of a matrix bias, the application of a ReLU
+// activation and the calculation of the maximum of the absolute value are
+// supported.
 class GraphString {
  public:
-  GraphString() : size_(0) {}
+  GraphString() = default;
 
   void AppendOp(std::string op_name, HloInstruction* op,
                 std::vector<HloInstruction*> operands = {}) {
-    graph_.append(
-        std::to_string(op->unique_id()) + ":[" +
-        primitive_util::LowercasePrimitiveTypeName(op->shape().element_type()) +
-        "]" + op_name + "(");
+    std::optional<int64_t> operand_uid;
     for (int i = 0; i < operands.size(); ++i) {
-      graph_.append(std::to_string(operands[i]->unique_id()));
-      if (i < operands.size() - 1) {
-        graph_.append(",");
+      if (OpInGraph(operands[i]->unique_id())) {
+        operand_uid = operands[i]->unique_id();
```

Comment:
It's possible that both operands are in the graph, e.g. if an Add instruction has the same operand twice. Handle this case. One way is to return a bool, and have the caller, CaptureConvGraphRecursive, give up in this case. Or DCHECK that only one operand is in the graph and have the caller check if both are in the graph or not.